core/Cargo.toml

```diff
@@ -104,6 +104,7 @@ http = "0.2.5"
 hyper = "0.14"
 lazy-regex = { version = "2.5.0", optional = true }
 log = "0.4"
+madsim = "0.2.19"
```

Comment:
It's better to include `madsim` as `optional` dep. We can add a new feature called `layers-madsim` for it.